resources/adoptopenjdk_install.rb

```diff
@@ -0,0 +1,178 @@
+#
+# Cookbook:: java
+# Resource:: adoptopenjdk_install
+#
+# Based on oracle_install.rb by Bryan W. Berry (<bryan.berry@gmail.com>)
+#
+
+resource_name :adoptopenjdk_install
+
+default_action :install
+
+property :url, String
+property :checksum, String, regex: /^[0-9a-f]{32}$|^[a-zA-Z0-9]{40,64}$/
+property :md5, String, regex: /^[0-9a-f]{32}$|^[a-zA-Z0-9]{40,64}$/
+property :app_home, String
+property :app_home_mode, Integer, default: 0755
+property :bin_cmds, Array, default: []
+property :owner, String, default: 'root'
+property :group, String, default: lazy { node['root_group'] }
+property :default, [true, false], default: true
+property :alternatives_priority, Integer, default: 1
+property :reset_alternatives, [true, false], default: true
+property :variant, ['hotspot', 'openj9', 'openj9-large-heap'], default: 'openj9'
+
+action :install do
+  app_dir_name, tarball_name = parse_app_dir_name(new_resource.url)
+  app_root = new_resource.app_home.split('/')[0..-2].join('/')
+  app_dir = app_root + '/' + app_dir_name + '-' + new_resource.variant
+  app_group = new_resource.group
+  app_home = new_resource.app_home
+  Chef::Log.debug("processing #{new_resource.variant} variant")
+
+  directory app_root do
+    owner new_resource.owner
+    group app_group
+    mode new_resource.app_home_mode
+    recursive true
+    action :nothing
+  end.run_action(:create)
+
+  unless ::File.exist?(app_dir)
+    download_path = "#{Chef::Config[:file_cache_path]}/#{tarball_name}"
+    if adoptopendjk_downloaded?(download_path, new_resource)
+      Chef::Log.debug('AdoptOpenJDK tarball already downloaded, not downloading again')
+    else
+      Chef::Log.debug("downloading tarball from #{URI.parse(new_resource.url).host}")
+      remote_file "#{Chef::Config[:file_cache_path]}/#{tarball_name}" do
+        source new_resource.url
+        checksum new_resource.checksum
+        retries new_resource.retries
+        retry_delay new_resource.retry_delay
+        mode '0755'
```

Comment:
I am assuming you are doing this to account for more restrictive umask, does this really need word execute?  Chef will be executing this in 99.999% of all cases as root so it really does not matter. Not that I see an issue with this really as you are trusting the source just trying to understand if there is a specific reason I am not recognizing.

Reply:
I corrected the mode to be 644. Thanks!